qopt_best_practices/transpilation/swap_cancellation_pass.py

```diff
@@ -0,0 +1,44 @@
+"""Pass to remove SWAP gates that are not needed."""
+
+from qiskit.dagcircuit import DAGOutNode, DAGCircuit
+from qiskit.transpiler import TransformationPass
+
+
+class SwapToFinalMapping(TransformationPass):
+    """Absorb any redundent SWAPs in the final layout.
+
+    This pass should be executed after a SWAPStrategy has been applied to a block
+    of commuting gates. It will remove any final redundent SWAP gates and absorb
+    them into the virtual layout. This effectively undoes any possibly redundent
+    SWAP gates that the SWAPStrategy may have inserted.
+    """
+
+    def run(self, dag: DAGCircuit):
+        """run the pass."""
+
+        qmap = self.property_set["virtual_permutation_layout"]
+
+        qreg = dag.qregs["q"]
```

Comment:
I know the code isn't new, but I think you can try to find a more resilient way to access the qreg, this line would fail if someone happens to give the registers a custom name. What about
```suggestion
        qreg = dag.qregs[next(iter(dag.qregs))]
```